PurchasesTests/Attribution/AttributionPosterTests.swift

```diff
@@ -26,7 +26,13 @@ class AttributionPosterTests: XCTestCase {
     var deviceCache: MockDeviceCache!
     var identityManager: MockIdentityManager!
     var backend: MockBackend!
-    let subscriberAttributesManager = MockSubscriberAttributesManager()
```

Comment:
i don't understand how this was working without passing parameters before?

Reply:
it was using the `NSObject -init` which doesn't take params 🤦

Reply:
yeah, NSObjects will have an implicit `init` with no params, but if you declare at least one `init` with params in Swift, then the logic that converts swift into obj-c will declare the `init` with no params as `unavailable` and the one with params as the `designated` init. 

So until you have an init with params, calling it with no params will work. Once you add one, the old calls will break.

Reply:
okay so the init with params in the parent class doesn't count towards that rule i guess

Reply:
Hah, nope. I mean, why would it? That would make way too much sense.

Reply:
LOL

The one in the parent class doesn't count towards that rule because the automatic porting is doing a bit more than what we did in the parent class: 

- declares the `init` initializer as `NS_UNAVAILABLE`
- declares the other init as `DESIGNATED_INITIALIZER`

If we'd done that too (which... we should have), we'd have the same result as what we get through the automatic porting